File: app/src/main/java/com/wavesplatform/wallet/request/OrderType.java
package com.wavesplatform.wallet.request;

public enum OrderType {
    buy(0, "buy"),
    sell(1, "sell");

    private String name;
    private int value;

    OrderType(int value, String name) {
        this.value = value;
        this.name = name;
    }

    public byte[] toBytes() {
        return new byte[]{(byte) value};
    }

    public static OrderType fromString(String text) {
        if (buy.name.compareToIgnoreCase(text) == 0) {
            return buy;
        } else if (sell.name.compareToIgnoreCase(text) == 0) {
            return sell;
        }

        return null;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: app/src/main/java/com/wavesplatform/wallet/v2/data/model/local/OrderType.kt
package com.wavesplatform.wallet.v2.data.model.local

import com.google.gson.annotations.SerializedName
import com.wavesplatform.wallet.R
import pers.victor.ext.findColor

enum class OrderType(private val typeServerByte: Int, val typeUI: Int, val color: Int) {
    @SerializedName("buy")
    BUY(0, R.string.my_orders_type_buy, findColor(R.color.submit400)),
    @SerializedName("sell")
    SELL(1, R.string.my_orders_type_sell, findColor(R.color.error400));

    fun toBytes(): ByteArray {
        return byteArrayOf(typeServerByte.toByte())
    }
}
